src/components/LmsConfigurations/DegreedIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,292 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+import { handleErrors, validateLmsConfigForm } from './common';
+
+export const REQUIRED_DEGREED_CONFIG_FIELDS = [
+  'degreedBaseUrl',
+  'degreedUserId',
+  'degreedUserPassword',
+  'secret',
+  'degreedCompanyId',
+  'key',
+];
+
+function DegreedIntegrationConfigForm({ enterpriseId, config }) {
+  const [invalidFields, setInvalidFields] = useState({});
```

Comment:
These names really highlight how we reuse names elsewhere (I had to change what invalidFields somewhere else to something else. :()